app/controllers/user_confirmations_controller.rb

```diff
@@ -36,6 +36,10 @@ def create
   def show
     self.resource = resource_class.confirm_by_token(params[:confirmation_token])
 
+    Spree::Order.placed_by_customer_email(resource.email).each do |order|
```

Comment:
Are you sure that, at this point, user is pretty well confirmed?

Reply:
Uh, this is getting a bit complicated. Don't we do this already? No, I found two similar actions:

* When a user is created, we associate all customer records by email.
* When an order is validated, we associate a customer record.

Now, I think, we want to find guest orders when a customer record is created. That should happen in the model, possibly when building the customer record (before saving).

A view is not the right place to modify records. It leaves the database in an inconsistent state if nobody looks at that page and it does it too often if you look at the page several times.

We could do it in a background job but it's not an expensive operation. I think that saving this with a new customer record would be best. Once the customer record is there, you can't check out as guest with the same email.

Reply:
> Now, I think, we want to find guest orders when a customer record is created.

Not sure to understand your point here.

Shouldn't we do this after a user is created?